### ISSUE TITLE
Wait containers stuck in pod watching 403

### ISSUE THREAD

**Issue by terrytangyuan** [argoproj/argo-workflows#5371]
## Summary 

What happened/what you expected to happen?

```
k3d cluster create k3s-default
k create namespace argo
k apply -n argo -f manifests/install.yaml
k edit configmap -n argo workflow-controller-configmap # update configmap to use k8sapi
k create -f https://raw.githubusercontent.com/argoproj/argo-workflows/master/examples/resubmit.yaml
```

```
> k get pods -n argo
NAME                                   READY   STATUS     RESTARTS   AGE
workflow-controller-64c8b7c5b7-dtlzh   1/1     Running    0          14m
argo-server-7fd688c895-sm7x6           1/1     Running    1          14m
resubmit-d8knp-765531043               1/2     NotReady   0          9m54s
resubmit-d8knp-3592006595              1/2     NotReady   0          9m54s
resubmit-d8knp-782308662               1/2     Error      0          9m54s

> k logs -n argo resubmit-d8knp-3592006595 main
exiting with code 0

> k logs -n argo resubmit-d8knp-3592006595 wait
time="2021-03-11T20:17:07.925Z" level=info msg="Watch pods 403"
time="2021-03-11T20:17:07.926Z" level=info msg="Watch pods 403"
time="2021-03-11T20:17:07.928Z" level=info msg="Watch pods 403"
time="2021-03-11T20:17:07.928Z" level=info msg="Watch pods 403"

> k logs -n argo resubmit-d8knp-782308662 main
exiting with code 1

> k logs -n argo resubmit-d8knp-782308662 wait
time="2021-03-11T20:17:07.925Z" level=info msg="Watch pods 403"
time="2021-03-11T20:17:07.926Z" level=info msg="Watch pods 403"
time="2021-03-11T20:17:07.928Z" level=info msg="Watch pods 403"
time="2021-03-11T20:17:07.928Z" level=info msg="Watch pods 403"
```

## Diagnostics

What Kubernetes provider are you using? 

```
k3d version v3.4.0
k3s version latest (default)
```

What version of Argo Workflows are you running? 

v3.0.0-rc6 or latest from master branch with k8sapi

```
Paste the logs from the workflow controller:
kubectl logs -n argo $(kubectl get pods -l app=workflow-controller -n argo -o name) | grep ${workflow}
```

```
time="2021-03-11T20:05:56.707Z" level=info msg="Detected ConfigMap update."
time="2021-03-11T20:05:56.707Z" level=info msg="Configuration:\nartifactRepository: {}\ncontainerRuntimeExecutor: k8sapi\ninitialDelay: 0s\nmetricsConfig: {}\nnodeEvents: {}\npodSpecLogStrategy: {}\ntelemetryConfig: {}\n"
time="2021-03-11T20:05:56.707Z" level=info msg="Persistence configuration disabled"
time="2021-03-11T20:05:58.539Z" level=info msg="Get leases 200"
time="2021-03-11T20:05:58.542Z" level=info msg="Update leases 200"
time="2021-03-11T20:06:03.545Z" level=info msg="Get leases 200"
time="2021-03-11T20:06:03.549Z" level=info msg="Update leases 200"
time="2021-03-11T20:06:08.551Z" level=info msg="Get leases 200"
time="2021-03-11T20:06:08.556Z" level=info msg="Update leases 200"
time="2021-03-11T20:06:13.559Z" level=info msg="Get leases 200"
time="2021-03-11T20:06:13.563Z" level=info msg="Update leases 200"
time="2021-03-11T20:06:18.574Z" level=info msg="Get leases 200"
time="2021-03-11T20:06:18.579Z" level=info msg="Update leases 200"
time="2021-03-11T20:06:23.583Z" level=info msg="Get leases 200"
time="2021-03-11T20:06:23.587Z" level=info msg="Update leases 200"
time="2021-03-11T20:06:28.591Z" level=info msg="Get leases 200"
time="2021-03-11T20:06:28.596Z" level=info msg="Update leases 200"
time="2021-03-11T20:06:33.600Z" level=info msg="Get leases 200"
time="2021-03-11T20:06:33.604Z" level=info msg="Update leases 200"
time="2021-03-11T20:06:38.607Z" level=info msg="Get leases 200"
time="2021-03-11T20:06:38.612Z" level=info msg="Update leases 200"
time="2021-03-11T20:06:40.438Z" level=info msg="Processing workflow" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.440Z" level=info msg="Get configmaps 404"
time="2021-03-11T20:06:40.441Z" level=info msg="resolved artifact repository" artifactRepositoryRef=default-artifact-repository
time="2021-03-11T20:06:40.441Z" level=info msg="Updated phase  -> Running" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.444Z" level=info msg="DAG node resubmit-d8knp initialized Running" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.446Z" level=info msg="All of node resubmit-d8knp.B dependencies [] completed" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.447Z" level=info msg="Steps node resubmit-d8knp-3608784214 initialized Running" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.447Z" level=info msg="StepGroup node resubmit-d8knp-1253871432 initialized Running" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.448Z" level=info msg="Pod node resubmit-d8knp-765531043 initialized Pending" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.457Z" level=info msg="Create events 201"
time="2021-03-11T20:06:40.471Z" level=info msg="Create pods 201"
time="2021-03-11T20:06:40.475Z" level=info msg="Created pod: resubmit-d8knp.B[0].randfail1a (resubmit-d8knp-765531043)" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.476Z" level=info msg="Pod node resubmit-d8knp-782308662 initialized Pending" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.483Z" level=info msg="Create pods 201"
time="2021-03-11T20:06:40.484Z" level=info msg="Created pod: resubmit-d8knp.B[0].randfail1b (resubmit-d8knp-782308662)" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.484Z" level=info msg="Workflow step group node resubmit-d8knp-1253871432 not yet completed" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.484Z" level=info msg="All of node resubmit-d8knp.A dependencies [] completed" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.485Z" level=info msg="Pod node resubmit-d8knp-3592006595 initialized Pending" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.504Z" level=info msg="Create pods 201"
time="2021-03-11T20:06:40.507Z" level=info msg="Created pod: resubmit-d8knp.A (resubmit-d8knp-3592006595)" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:40.538Z" level=info msg="Update workflows 200"
time="2021-03-11T20:06:40.538Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=943 workflow=resubmit-d8knp
time="2021-03-11T20:06:43.616Z" level=info msg="Get leases 200"
time="2021-03-11T20:06:43.621Z" level=info msg="Update leases 200"
time="2021-03-11T20:06:48.630Z" level=info msg="Get leases 200"
time="2021-03-11T20:06:48.650Z" level=info msg="Update leases 200"
time="2021-03-11T20:06:50.472Z" level=info msg="Processing workflow" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:50.473Z" level=info msg="Updating node resubmit-d8knp-3592006595 message: ContainerCreating" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:50.474Z" level=info msg="Updating node resubmit-d8knp-765531043 message: ContainerCreating" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:50.474Z" level=info msg="Updating node resubmit-d8knp-782308662 message: ContainerCreating" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:50.477Z" level=info msg="Workflow step group node resubmit-d8knp-1253871432 not yet completed" namespace=argo workflow=resubmit-d8knp
time="2021-03-11T20:06:50.506Z" level=info msg="Update workflows 200"
time="2021-03-11T20:06:50.506Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=962 workflow=resubmit-d8knp
time="2021-03-11T20:06:52.751Z" level=info msg="Alloc=11422 TotalAlloc=34438 Sys=73297 NumGC=9 Goroutines=201"
time="2021-03-11T20:06:53.654Z" level=info msg="Get leases 200"
time="2021-03-11T20:06:53.658Z" level=info msg="Update leases 200"
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by alexec** [comment]
Does your SA have ‘watch pod’ as well as ‘get pod’?

**Comment by terrytangyuan** [comment]
I applied https://github.com/argoproj/argo-workflows/blob/master/manifests/install.yaml directly with a fresh namespace and k3d cluster.

**Comment by terrytangyuan** [comment]
> Does your SA have ‘watch pod’ as well as ‘get pod’?

Actually good reminder. I added `serviceAccountName: argo` and then everything works as expected now.

**Comment by terrytangyuan** [comment]
On the other hand, emissary executor has better logs `pods "resubmit-qpft5-2434376666" is forbidden: User "system:serviceaccount:argo:default" cannot patch resource "pods" in API group "" in the namespace "argo"`

**Comment by alexec** [comment]
Would you like to submit PR to improve messaging?

**Comment by terrytangyuan** [comment]
Sure

**Comment by simster7** [comment]
This has been fixed in [`v3.0.0-rc9`](https://github.com/argoproj/argo-workflows/releases/tag/v3.0.0-rc9)
